test/utils.sh

```diff
@@ -1199,3 +1199,40 @@ retry() {
       sleep "${interval}"
   done
 }
+
+# This function will be used by tasks in tekton-integration-catalog
+# If the image is an index (multi-arch), replace the digest with the 0th manifest's digest; if it's a single-arch image manifest, leave it unchanged.
+resolve_arch_specific_digest() {
+  local FBC_FRAGMENT_IMAGE="$1"
+
+  # Validate input parameters
+  if [[ -z "$FBC_FRAGMENT_IMAGE" ]]; then
+    echo "resolve_arch_specific_digest: Invalid input. Usage: resolve_arch_specific_digest <FBC_FRAGMENT_IMAGE>" >&2
+    exit 2
+  fi
+
+  # Inspect the raw image data
+  local raw_output
+  if ! raw_output=$(retry skopeo inspect --raw "docker://${FBC_FRAGMENT_IMAGE}"); then
+    echo "resolve_arch_specific_digest: Failed to inspect the image: ${FBC_FRAGMENT_IMAGE}" >&2
+    exit 1
+  fi
+
+  # Get manifest count (number of platform-specific variants)
+  local manifest_count
+  manifest_count=$(echo "$raw_output" | jq '.manifests | length')
+
+  if [[ "$manifest_count" -gt 1 ]]; then
+    # Extract the digest of the 0th manifest
+    local new_digest
+    new_digest=$(echo "$raw_output" | jq -r '.manifests[0].digest')
+
+    # Replace the digest in the original FBC fragment
+    local updated_fragment
+    updated_fragment=$(echo "$FBC_FRAGMENT_IMAGE" | sed -E 's/@sha256:[a-f0-9]+/@'"$new_digest"'/')
```

Comment:
We can use a function like `get_image_registry_repository_tag` to trim off the digest and then append the new one. The exact function you would want to use depends on if you want the tag to be there as well.